.gitmodules

```diff
@@ -1,3 +1,3 @@
 [submodule "linux"]
 	path = linux
-	url = ssh://git@github.com/Ultimaker/linux.git
+	url = ../linux
```

Comment:
This will brake local building and i am surprised that it works remotely we should not rely on a relative paths.

Reply:
As discussed yesterday, I think we should for the following reasons:

1. It's recommended in the [Gitlab manual](https://docs.gitlab.com/ee/ci/git_submodules.html#using-git-submodules-in-your-ci-jobs).
2. Authentication will become an issue in the future;
   - Gitlab does not support authenticating over SSH.
   - Gitlab only supports cloning over HTTPS _without authentication_. This means that if we decide to keep our source code more secure in the future, CI (and probably other things) will break.

Furthermore, I think it's reasonable to expect a developer to clone the linux repo locally so Git can access it.

I propose improving upon this for EMP-645, but it could be done in this ticket as well if needed.

Reply:
This should work fine, we use this structure in all cloud projects as well

Reply:
As discussed, this is problematic when building locally. We always need the shared repository, a local relative clone can easily be out of sync.

Reply:
Git will use a relative sync to your origin's path, not the local file system...

Reply:
Ok i stand corrected if this works, and this is how it should be implemented and it does not cause out of sync problems (i will try to read up myself later), than sorry for the hussle Martijn.

Reply:
Did some extra digging and found this in the [git manual](https://git-scm.com/docs/git-submodule):

\<repository\> is the URL of the new submodule’s origin repository. **This may be either an absolute URL, or (if it begins with ./ or ../), the location relative to the superproject’s default \>_remote_\< repository**

Seems like we both misunderstood... it doesn't use local files, so there is no problem :) .

Reply:
I didn't refresh the page :D .